tests/upgrades/test_performance_tuning.py

```diff
@@ -0,0 +1,153 @@
+                            "\n# Do note, setting some values may have unintended " \
+                            "consequences that affect the" \
+                            "\n# performance or functionality of the application. " \
+                            "Consider the impact of your" \
+                            "\n# changes before applying them, and test them in a " \
+                            "non-production environment" \
+                            "\n# first." \
+                            "\n#" \
+                            "\n# Here are some examples of how you tune the Apache " \
+                            "options if needed:" \
+                            "\n#" \
+                            "\n# apache::mod::prefork::startservers: 8" \
+                            "\n# apache::mod::prefork::minspareservers: 5" \
+                            "\n# apache::mod::prefork::maxspareservers: 20" \
+                            "\n# apache::mod::prefork::serverlimit: 256" \
+                            "\n# apache::mod::prefork::maxclients: 256" \
+                            "\n# apache::mod::prefork::maxrequestsperchild: 4000" \
+                            "\n# Here are some examples of how you tune the " \
+                            "PostgreSQL options if needed:" \
+                            "\n#" \
+                            "\n# postgresql::server::config_entries:" \
+                            "\n#   max_connections: 600" \
+                            "\n#   shared_buffers: 1024MB" \
+                            "\n# Do not remove " \
+                            "\nmongodb::server::storage_engine: 'wiredTiger'"
+
+
+class ScenarioPerfornaceTuning(TestCase):
+    """The test class contains pre-upgrade and post-upgrade scenarios to test
+    Performance Tuning utility
+
+    Test Steps:
+
+    1. Before Satellite upgrade, we will apply the medium tune size.
+    2. Once the size get apply we will wait for post upgrade test case to check
+    whether the size is same or not.
+    3. In post upgrade we will check whether satellite installer would able to change
+    the size or not
+
+    """
+
+    @pre_upgrade
+    def test_pre_performance_tuning_apply(self):
+        """In preupgrade scenario we will apply the medium tuning size.
+
+        :id: preupgrade-83404326-20b7-11ea-a370-48f17f1fc2e1
+
+        :steps:
+            1. Run satellite-installer --tuning medium --disable-system-checks.
+
+        :expectedresults: Medium tuning parameter should be applied.
+
+         """
+        with open('custom-hiera.yaml', 'w') as fd:
+            fd.write("{}\n".format(DEFAULT_CUSTOM_HIERA_DATA))
+            fd.write("{}".format(MEDIUM_TUNING_DATA))
+        ssh.upload_file('custom-hiera.yaml', '/etc/foreman-installer')
+        command_output = [data.strip(" ") for data in
+                          ssh.command('satellite-installer --disable-system-checks')
+                              .stdout]
```

Comment:
I think it would be more readable if you had `ssh` call outside of comprehension:
```python
command_output = ssh.command('satellite-installer --disable-system-checks').stdout
command_output = [data.strip(" ") for data in command_output]
```

Reply:
fixed.